internal/provider/resource_host_catalog_plugin.go

```diff
@@ -0,0 +1,413 @@
+		ReadContext:   resourceHostCatalogPluginRead,
+		UpdateContext: resourceHostCatalogPluginUpdate,
+		DeleteContext: resourceHostCatalogPluginDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			IDKey: {
+				Description: "The ID of the host catalog.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			NameKey: {
+				Description: "The host catalog name. Defaults to the resource name.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			DescriptionKey: {
+				Description: "The host catalog description.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			ScopeIdKey: {
+				Description: "The scope ID in which the resource is created.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			PluginIdKey: {
+				Description: "The ID of the plugin that should back the resource. This or " + PluginNameKey + " must be defined.",
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Computed:    true, // If name is provided this will be computed
+			},
+			PluginNameKey: {
+				Description: "The name of the plugin that should back the resource. This or " + PluginIdKey + " must be defined.",
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Computed:    true,
+			},
+			TypeKey: {
+				Description: "The host catalog type. Only `plugin` is supported, and is the default.",
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Optional:    true,
+				Default:     "plugin",
+			},
+			AttributesJsonKey: {
+				Description: `The attributes for the host catalog. Either values encoded with the "jsonencode" function, pre-escaped JSON string, or a file:// or env:// path. Set to a string "null" or remove the block to clear all attributes in the host catalog.`,
+				Type:        schema.TypeString,
+				Optional:    true,
+				// If set to null in config and nothing comes from API, consider
+				// it the same. Same if config changes from empty to null.
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					switch {
+					case old == new:
+						return true
+					case old == "null" && new == "":
+						return true
+					case old == "" && new == "null":
+						return true
+					default:
+						return false
+					}
+				},
+			},
+			SecretsJsonKey: {
+				Description: `The secrets for the host catalog. Either values encoded with the "jsonencode" function, pre-escaped JSON string, or a file:// or env:// path. Set to a string "null" to clear any existing values. NOTE: Unlike "attributes_json", removing this block will NOT clear secrets from the host catalog; this allows injecting secrets for one call, then removing them for storage.`,
+				Type:        schema.TypeString,
+				Optional:    true,
+				Sensitive:   true,
+			},
+			SecretsHmacKey: {
```

Comment:
For my education, I don't see how the hmac is being used here.  Can you point me to how it is used?

Reply:
Right now, it's only used in tests -- it lets us verify that secret creating, updating, and nulling are working properly.

In theory it can be used to raise an issue to the user if and when a read secrets hmac value does not match state, indicating it was changed outside of TF's purview. However, we wouldn't want to just blindly fail TF runs so I think we'd need some nuance there, and I didn't want to tackle that yet.

That said, it is _very_ useful for tests, as it'd be impossible to validate functionality without it.